src/services/EdFi.DataManagementService.Api/Core/Validation/DocumentValidator.cs

```diff
@@ -0,0 +1,76 @@
+// SPDX-License-Identifier: Apache-2.0
+// Licensed to the Ed-Fi Alliance under one or more agreements.
+// The Ed-Fi Alliance licenses this file to you under the Apache License, Version 2.0.
+// See the LICENSE and NOTICES files in the project root for more information.
+
+using System.Text.Json.Nodes;
+using EdFi.DataManagementService.Api.ApiSchema.Extensions;
+using Json.Schema;
+
+namespace EdFi.DataManagementService.Api.Core.Validation;
+
+public interface IDocumentValidator
+{
+    /// <summary>
+    /// Validates input json
+    /// </summary>
+    /// <param name="input"></param>
+    /// <param name="validatorContext"></param>
+    /// <returns></returns>
+    IEnumerable<string>? Validate(JsonNode? input, ValidatorContext validatorContext);
+}
+
+public class DocumentValidator(ISchemaValidator schemaValidator) : IDocumentValidator
+{
+    public IEnumerable<string>? Validate(JsonNode? input, ValidatorContext validatorContext)
```

Comment:
Summary doc on what this does e.g. validates a document body against a JSON Schema. Also suggest renaming `input` to more descriptive `document`, `body`, `documentBody`, etc.